lib/repositories/message.js

```diff
@@ -394,20 +412,89 @@ function createApi(table, provider, sqlGenerator, dependencies) {
                 });
             });
         });
+    },
+
+    /**
+     * Deletes all messages belonging to the specific mailbox
+     */
+    removeByMailbox: function remove(mailbox) {
```

Comment:
Since you name the function removeByMailbox, I would remove the remove name after function. Some people encourage this so that stack traces can show the name of the function where an error occured. In that case however, it should read removeByMailbox: function removeByMailbox(mailbox).

I don't personally like the duplication of names so I don't bother with it.